### PR TITLE
Refactor  and add new functions to util.py

### PR DESCRIPTION
**Pull request by erogluorhan**
This PR performs the following changes:

1) Name of function `nclize_axis()` has been changed to `add_major_minor_ticks()`

2) `add_major_minor_ticks(ax, minor_per_major=3) `function signature is modified as `add_major_minor_ticks(ax, x_minor_per_major=3, y_minor_per_major=3, labelsize="small")` in order to control label size explicitly as well as number of major and minor ticks differently from each other.

3) A new utility function has been added as follows in order to provide handling of axes titles, left/right titles, and labels from within only one function:

       `set_titles_and_labels(ax, title=None, titlefontsize=18, left=None, leftfontsize=18, right=None, rightfontsize=18, xlabel=None, ylabel=None, labelfontsize=16)`

4) A new utility function has been added as follows in order to provide handling of axes titles, left/right titles, and labels from within only one function:

       `set_axes_limits_and_ticks(ax, xlim=None, ylim=None, xticks=None, yticks=None, xticklabels=None, yticklabels=None)`

5) A new utility function has been added as follows in order to handle the no-shown-data artifact of 0 and 360-degree longitudes:

       `xr_add_cyclic(da, coord)`

        **This function adds dependency for xarray which has been also added under `setup.py`.**

**IMPORTANT: This PR will break most of the geocat-examples function that have been already implemented. They are going to be resolved as soon as this PR is merged.**

### CODE REVIEW

**Comment by mnlevy1981** [comment]
As mentioned in https://github.com/NCAR/GeoCAT-examples/pull/52#issuecomment-584839695 I would recommend pinning the old version of this repo in the [conda_environment.yml](https://github.com/NCAR/GeoCAT-examples/blob/master/conda_environment.yml) file for [GeoCAT-examples](https://github.com/NCAR/GeoCAT-examples) before accepting this PR.

File: src/util.py [hunk 1]
```diff
@@ -74,21 +228,27 @@ def truncate_colormap(cmap, minval=0.0, maxval=1.0, n=100, name=None):
...
-    warnings.filters.pop(0)
+    import xarray as xr
+    import cartopy.util as cutil
+
+    cyclic_data, cyclic_coord = cutil.add_cyclic_point(da.values, coord=da[coord])
+
+    coords = da.coords.to_dataset()
+    coords[coord] = cyclic_coord
+
+    return xr.DataArray(cyclic_data, dims=da.dims, coords=coords.coords, attrs=da.attrs, encoding=da.encoding)
```
Comment (khallock) [inline src/util.py]: A quick heads up, I found an issue with this `encoding` keyword argument being deprecated in Xarray as of version 0.14 and removed in 0.15. See NCAR/GeoCAT-examples#55 for a possible fix.

File: src/util.py [hunk 2]
```diff
@@ -25,34 +37,78 @@ def nclize_axis(ax, minor_per_major=3):
...
+                  'deprecated, please use geocat.viz.cmaps.BlueYellowRed '
+                  'instead.', DeprecationWarning, stacklevel=2)
+    warnings.filters.pop(0)
+
+    return cmaps.BlueYellowRed
+
+def set_titles_and_labels(ax, title=None, titlefontsize=18, left=None, leftfontsize=18, right=None, rightfontsize=18,
+                          xlabel=None, ylabel=None, labelfontsize=16):
+    """
+    Utility function to handle axis titles, left/right titles, and labels
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply src/util.py]: Yes, that's correct. It's now addressed with new commits.

File: src/util.py [hunk 3]
```diff
@@ -25,34 +37,78 @@ def nclize_axis(ax, minor_per_major=3):
...
+
+    if xlabel is not None:
+        ax.set_xlabel(xlabel, fontsize=labelfontsize)
+
+    if ylabel is not None:
+        ax.set_ylabel(ylabel, fontsize=labelfontsize)
+
+def set_axes_limits_and_ticks(ax, xlim=None, ylim=None, xticks=None, yticks=None, xticklabels=None, yticklabels=None):
+    """
+    Utility function to determine axis limits, tick values and labels
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply src/util.py]: This is now addressed with new commits.

[the remark this answers is not in the document]
Comment (erogluorhan) [reply src/util.py]: This has been addressed with parameter docstrings.

**Pull request merged**